File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.ts
function findMinArrowShots(points: number[][]): number {
    if(!points.length) return 0
    const pointsSorted = points.sort((a, b) => a[0] === b[0] ? b[1]-a[1] : a[0] - b[0])
    let totalShot = 1

    let left = points[0][0], right = points[0][1]
    for(let [start, end] of points){
        if(left>end || right < start){
            left = start
            right = end
            totalShot++
        }else{
            left = Math.max(left, start)
            right = Math.min(right, end)
        }
    }

    return totalShot
};

File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key = lambda x: (x[0], -x[1]))
        
        left, right = points[0][0], points[0][1]

        shots = 1
        for start, end in points:
            left = max(left, start)
            right = min(right, end)

            if left > right:
                shots += 1
                left, right = start, end


        # print(points)
        return shots
        